src/librustc/ich/hcx.rs

```diff
@@ -320,7 +320,7 @@ impl<'a> HashStable<StableHashingContext<'a>> for Span {
         const TAG_EXPANSION: u8 = 0;
         const TAG_NO_EXPANSION: u8 = 1;
 
-        if !hcx.hash_spans {
+        if !hcx.hash_spans && false {
```

Comment:
I believe this should be 

```suggestion
        if !hcx.hash_spans || true {
```
Right now it doesn't ever do an early-return.

Reply:
Ah, crap `:)` That's what you get when you do a PR right before going home...